### PR TITLE
[Refactor] Add backend manager for 1.x

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

backend  manager for 1.x https://github.com/open-mmlab/mmdeploy/pull/1475

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by lvhan028 — APPROVED** [review]

File: docs/en/07-developer-guide/support_new_backend.md [hunk 1]
```diff
@@ -123,32 +123,20 @@ The backends in MMDeploy must support the ONNX. The backend loads the ".onnx" fi
        __all__ += ['onnx2ncnn', 'get_output_model_file']
    ```
 
-   Then add the codes about conversion to `tools/deploy.py` using these APIs if necessary.
+   Create a backend manager class which derive from `BackendManager`, implement its `to_backend` static method.
```
Comment (AllentDan) [inline docs/en/07-developer-guide/support_new_backend.md:126]: Does `BackendManager` mean `BaseBackendManager`?

File: docs/en/07-developer-guide/support_new_backend.md [hunk 2]
```diff
@@ -209,23 +197,26 @@ Although the backend engines are usually implemented in C/C++, it is convenient
...
-                          input_names: Optional[Sequence[str]] = None,
-                          output_names: Optional[Sequence[str]] = None):
-           if backend == Backend.ONNXRUNTIME:
-               from mmdeploy.backend.onnxruntime import ORTWrapper
-               return ORTWrapper(
-                   onnx_file=backend_files[0],
-                   device=device,
-                   output_names=output_names)
+        @BACKEND_MANAGERS.register('onnxruntime')
+        class ONNXRuntimeUtils(BaseBackendManager):
```
Comment (AllentDan) [inline docs/en/07-developer-guide/support_new_backend.md:206]: Utils?

File: mmdeploy/backend/tvm/backend_manager.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                   work_dir: str,
+                   deploy_cfg: Any,
+                   log_level: int = logging.INFO,
+                   device: str = 'cpu',
+                   **kwargs) -> Sequence[str]:
+        """Convert intermediate representation to given backend.
+
+        Args:
+            ir_files (Sequence[str]): The intermediate representation files.
+            work_dir (str): The work directory, backend files and logs should
```
Comment (AllentDan) [inline mmdeploy/backend/tvm/backend_manager.py:78]: saved

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**